src/main/java/org/openrewrite/java/testing/junit5/RemoveTryCatchBlocksFromUnitTests.java

```diff
@@ -53,48 +53,60 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
         return new RemoveTryCatchBlocksFromUnitsTestsVisitor();
     }
 
-    private static class RemoveTryCatchBlocksFromUnitsTestsVisitor extends JavaIsoVisitor<ExecutionContext> {
+    private static class RemoveTryCatchBlocksFromUnitsTestsVisitor extends JavaVisitor<ExecutionContext> {
         @Override
         public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration md, ExecutionContext ctx) {
+            if (md.getBody() == null) {
+                return md;
+            }
+
+            List<Statement> statements = md.getBody().getStatements();
+            if (statements.stream().noneMatch(s -> s instanceof J.Try)) {
+                return md;
+            }
+
             // check if method is Unit test
             List<J.Annotation> annotations = md.getAllAnnotations();
             if (annotations.stream().noneMatch(ann -> MATCHERS.stream().anyMatch(matcher -> matcher.matches(ann)))) {
                 return md;
             }
 
-            if (md.getBody() == null) {
-                return md;
-            }
             // find try catch block
-            List<Statement> statements = md.getBody().getStatements();
-            List<Statement> tryStatements = statements.stream().filter(s -> s instanceof J.Try).collect(Collectors.toList());
+            List<J.Try> tryStatements = statements.stream()
+                    .filter(s -> s instanceof J.Try)
+                    .map(s -> (J.Try) s)
+                    .filter(t -> t.getCatches().size() == 1)
+                    .collect(Collectors.toList());
             if (tryStatements.size() == 0) {
                 return md;
             }
 
-            for (Statement statement : tryStatements) {
-                J.Try tryBlock = (J.Try) statement;
-                J.Block tryBody = tryBlock.getBody();
-                List<J.Try.Catch> catch_ = tryBlock.getCatches();
-                for (J.Try.Catch c : catch_) {
-                    J.Block block = c.getBody();
-                    List<Statement> catchStatements = block.getStatements();
-                    if (catchStatements.stream().anyMatch(
-                            st -> st instanceof J.MethodInvocation && ASSERT_FAIL_MATCHER.matches((J.MethodInvocation) st))
-                    ) {
-                        // replace method body
-                        maybeRemoveImport("org.junit.Assert");
-                        maybeAddImport("org.junit.jupiter.api.Assertions");
-                        return JavaTemplate.builder("Assertions.assertDoesNotThrow(() -> #{any()} )")
-                                .imports("org.junit.jupiter.api.Assertions")
-                                .javaParser(JavaParser.fromJavaVersion().classpathFromResources(ctx, "junit-jupiter-api-5.9"))
-                                .build()
-                                .apply(getCursor(), md.getCoordinates().replaceBody(), tryBody);
-                    }
+            for (J.Try try_ : tryStatements) {
```

Comment:
Most of this code can then be moved down into the `visitTry()` method. Then you don't need any looping either.